src/dotnet/MsbuildProject.cs

```diff
@@ -88,30 +96,30 @@ public static MsbuildProject FromDirectory(string projectDirectory)
                 throw new GracefulException(CommonLocalizableStrings.CouldNotFindAnyProjectInDirectory, projectDirectory);
             }
 
-            var project = TryOpenProject(projectFile.FullName);
+            var project = TryOpenProject(collection, projectFile.FullName);
             if (project == null)
             {
                 throw new GracefulException(CommonLocalizableStrings.FoundInvalidProject, projectFile.FullName);
             }
 
-            return new MsbuildProject(project);
+            return new MsbuildProject(collection, project);
         }
 
         public int AddProjectToProjectReferences(string framework, IEnumerable<string> refs)
         {
             int numberOfAddedReferences = 0;
 
-            ProjectItemGroupElement itemGroup = Project.FindUniformOrCreateItemGroupWithCondition(ProjectItemElementType, framework);
+            ProjectItemGroupElement itemGroup = ProjectRoot.FindUniformOrCreateItemGroupWithCondition(ProjectItemElementType, framework);
             foreach (var @ref in refs.Select((r) => NormalizeSlashes(r)))
             {
-                if (Project.HasExistingItemWithCondition(framework, @ref))
+                if (ProjectRoot.HasExistingItemWithCondition(framework, @ref))
                 {
                     Reporter.Output.WriteLine(string.Format(CommonLocalizableStrings.ProjectAlreadyHasAreference, @ref));
                     continue;
                 }
 
                 numberOfAddedReferences++;
-                itemGroup.AppendChild(Project.CreateItemElement(ProjectItemElementType, @ref));
+                itemGroup.AppendChild(ProjectRoot.CreateItemElement(ProjectItemElementType, @ref));
```

Comment:
reiterating the above. Reading this feels like we should be adding stuff in a directory. CLI repo uses Root for the purpose of `root directory` a lot, creating the ambiguity.